sled-agent/src/bootstrap/agent.rs

```diff
@@ -185,21 +194,41 @@ pub struct Agent {
 
     global_zone_bootstrap_link_local_address: Ipv6Addr,
 
-    // We maintain the handle just to show ownership, but don't use it
-    // as the KeyManager task should run forever
+    /// We maintain the handle just to show ownership, but don't use it, as the
+    /// KeyManager task should run forever
     #[allow(unused)]
     key_manager_handle: JoinHandle<()>,
 
-    // We maintain a copy of the `StorageKeyRequester` so we can pass it through
-    // from the `HardwareManager` to the `StorageManager` when the `HardwareManger`
-    // gets recreated.
+    /// Used to inofrm key requesters that the KeyManager is ready to be used.
```

Comment:
```suggestion
    /// Used to inform key requesters that the KeyManager is ready to be used.
```

Reply:
Fixed in https://github.com/oxidecomputer/omicron/pull/3652#discussion_r1265970198